client/core/core.go

```diff
@@ -7511,7 +7530,13 @@ func (c *Core) handleConnectEvent(dc *dexConnection, status comms.ConnectionStat
 	topic := TopicDEXDisconnected
 	if status == comms.Connected {
 		topic = TopicDEXConnected
+		dc.lastConnect = time.Now()
 	} else {
+		if time.Since(dc.lastConnect) < wsAnomalyDuration {
```

Comment:
Since `lastConnect` has no mutex protection but is only used in `handleConnectEvent`, there is an assumption that `(*wsConn).setConnectionStatus` will never be called concurrently. Are we sure about that?

Reply:
> Are we sure about that?

Double-checked, and  I think we need mutex protection for `lastConnect` because `(*wsConn).setConnectionStatus` is called from `(*wsConn).keepAlive`. Added in https://github.com/decred/dcrdex/pull/2028/commits/5c8f10f8bf3f47007fc056d5900cf2231af7679f